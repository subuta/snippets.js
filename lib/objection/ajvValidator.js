import { build, format, snippets as s } from 'bld.js'

export default () => {
  const imports = s.import([
    ['objection', null, [
      'AjvValidator'
    ]]
  ])

  return build`
    ${imports}

    const validator = new AjvValidator({
      onCreateAjv: (ajv) => {},
      options: {
        allErrors: true,
        validateSchema: false,
        ownProperties: true,
        removeAdditional: true,
        useDefaults: true
      }
    })
    
    // force override ajv schema.
    export const setSchema = (schema = {}) => {
      const $id = schema.$id || schema.id
      if (!$id) return
      // remove pre-added schemas first.
      validator.ajv.removeSchema($id)
      // then add passed schema
      validator.ajv.addSchema(schema)
    }
    
    ${s.export('validator')}
  `
}