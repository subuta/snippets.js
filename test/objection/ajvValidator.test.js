/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import ajvValidator from 'lib/objection/ajvValidator'

describe('objection/ajvValidator', () => {
  it('should create objection ajvValidator', () => {
    expect(format(ajvValidator())).toMatchSnapshot()
  })
})
