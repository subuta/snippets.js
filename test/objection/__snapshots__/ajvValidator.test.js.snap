// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objection/ajvValidator should create objection ajvValidator 1`] = `
"import {AjvValidator} from 'objection'

const validator = new AjvValidator({
  onCreateAjv: (ajv) => {},
  options: {
    allErrors: true,
    validateSchema: false,
    ownProperties: true,
    removeAdditional: true,
    useDefaults: true
  }
})

export default validator
"
`;
